backend/fleet_management/tests/test_drives.py

```diff
@@ -0,0 +1,87 @@
+from datetime import date
+
+from django.contrib.auth.models import Group
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from fleet_management.constants import Groups
+from fleet_management.factories import (
+    DriveFactory,
+    PassengerFactory,
+    CarFactory,
+    ProjectFactory,
+    UserFactory,
+)
+from fleet_management.models import Drive
+
+
+class DrivesApiTest(APITestCase):
+    def setUp(self):
+        self.url = reverse("drives")
+        self.passengers = PassengerFactory.create_batch(size=10)
+        self.car = CarFactory()
+        self.project = ProjectFactory(country="UA")
+        self.driver = UserFactory()
+        self.driver_group = Group.objects.filter(name=Groups.Driver.name)
+        self.driver.groups.set(self.driver_group)
+
+        self.drives = DriveFactory.create_batch(
+            size=10,
+            driver=self.driver,
+            passengers=self.passengers,
+        )
+
+    def test_401_for_unlogged_user(self):
+        res = self.client.get(self.url)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_can_retrieve_only_my_drives(self):
+        new_driver = UserFactory()
+        new_driver.groups.set(self.driver_group)
+        my_drive_id = DriveFactory(driver=new_driver).id
+
+        self.client.force_login(new_driver)
+        res = self.client.get(self.url)
+        drives = res.data
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(drives), 1)
+        self.assertEqual(drives[0]["id"], my_drive_id)
```

Comment:
I've just re-reviewed  `test_can_retrieve_only_my_drives` and `test_get_all_drives` and it occurs, that `test_can_retrieve_only_my_drives` was using `self.driver` to login to the application and check that others drivers' drives are excluded, so I would suppose that we will have have something like
```
self.client.force_login(self.driver)
...
self.assertEqual(len(drives), len(self.drives))
for drive in drives:
    self.assertEqual(drive["driver"]["id'], self.driver.id) 
```

Reply:
I think I fixed it in the next commit.